internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -882,85 +883,37 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
-// Some string checks require equality, others partial equality, whereas the int checks require numeric comparison.
-//
-// Ironic's internal filter process requires that all the disks have to fully match the RootDeviceHints (and operation),
-// which is what this function does.
-//
-// This function also filters out disks that are not elegible for installation, as we already know those cannot be used.
 func (r *BMACReconciler) findInstallationDiskID(devices []aiv1beta1.HostDisk, hints *bmh_v1alpha1.RootDeviceHints) string {
 	if hints == nil {
 		return ""
 	}
 
-	for _, disk := range devices {
-
-		if !disk.InstallationEligibility.Eligible {
-			continue
-		}
-
-		if hints.DeviceName != "" && hints.DeviceName != disk.Path {
-			continue
-		}
-
-		if hints.HCTL != "" && hints.HCTL != disk.Hctl {
-			continue
-		}
-
-		if hints.Model != "" && !strings.Contains(disk.Model, hints.Model) {
-			continue
-		}
-
-		if hints.Vendor != "" && !strings.Contains(disk.Vendor, hints.Model) {
-			continue
-		}
-
-		if hints.SerialNumber != "" && hints.SerialNumber != disk.Serial {
-			continue
-		}
-
-		if hints.MinSizeGigabytes != 0 {
-			sizeGB := int(disk.SizeBytes / (1024 * 3))
-			if hints.MinSizeGigabytes < sizeGB {
-				continue
-			}
-		}
-
-		if hints.WWN != "" && hints.WWN != disk.Wwn {
-			continue
-		}
-
-		// No WWNWithExtension
-		// if hints.WWWithExtension != "" && hints.WWWithExtension != disk.Wwwithextension {
-		// 	return ""
-		// }
-
-		// No WWNNVendorExtension
-		// if hints.WWNVendorExtension != "" && hints.WWNVendorExtension != disk.WwnVendorextension {
-		// 	return ""
-		// }
-
-		switch {
-		case hints.Rotational == nil:
-		case *hints.Rotational:
-			if !strings.EqualFold(disk.DriveType, string(models.DriveTypeHDD)) {
-				continue
-			}
-		case !*hints.Rotational:
-			if strings.EqualFold(disk.DriveType, string(models.DriveTypeHDD)) {
-				continue
-			}
+	disks := make([]*models.Disk, len(devices))
+
+	for i, dev := range devices {
+		disks[i] = &models.Disk{
+			Bootable:                dev.Bootable,
+			ByID:                    dev.ByID,
+			ByPath:                  dev.ByPath,
+			DriveType:               models.DriveType(strings.ToUpper(dev.DriveType)),
+			Hctl:                    dev.Hctl,
+			ID:                      dev.ID,
+			InstallationEligibility: models.DiskInstallationEligibility{Eligible: dev.InstallationEligibility.Eligible},
+			IoPerf:                  &models.IoPerf{SyncDuration: dev.IoPerf.SyncDurationMilliseconds},
+			Model:                   dev.Model,
+			Name:                    dev.Name,
+			Path:                    dev.Path,
+			Serial:                  dev.Serial,
+			SizeBytes:               dev.SizeBytes,
+			Smart:                   dev.Smart,
+			Vendor:                  dev.Vendor,
+			Wwn:                     dev.Wwn,
 		}
+	}
 
-		return disk.ID
+	acceptable := hostutil.GetAcceptableDisksWithHints(disks, hints)
+	if len(acceptable) > 0 {
+		return acceptable[0].ID
```

Comment:
Seems reasonable to pick the first acceptable disk though I do question whether or not there might need to be consideration of which of the acceptable is "best"

Are there any other priorities under which we would like to select the acceptable disk, is the first always the best choice?
Should the list of acceptable disks be sorted any way to place the "most acceptable" first?